app/ts/components/pages/PersonalSign.tsx

```diff
@@ -155,17 +162,48 @@ function SignRequest({ personalSignRequestData, renameAddressCallBack }: SignReq
 }
 
 
+const CellElement = (param: { text: ComponentChildren }) => {
+	return <div class = 'log-cell' style = 'justify-content: right;'> <p class = 'paragraph' style = 'color: var(--subtitle-text-color)'> { param.text }</p></div>
+}
+
+export function Permit2ExtraDetails({ permit2 }: { permit2: PersonalSignRequestDataPermit2 }) {
+	return <>
+		<CellElement text = 'Chain: '/>
+	<	CellElement text = { getChainName(BigInt(permit2.message.domain.chainId)) }/>
+		<CellElement text = 'Nonce: '/>
+		<CellElement text = { permit2.message.message.details.nonce.toString(10) }/>
+		<CellElement text = 'Signature expires  in:'/>
+		<CellElement text = { <SomeTimeAgo priorTimestamp = { new Date(Number(permit2.message.message.sigDeadline) * 1000) } countBackwards = { true }/> }/>
+		<CellElement text = 'Spender can spend for:'/>
+		<CellElement text = { <>
+			<SomeTimeAgo priorTimestamp = { new Date(Number(permit2.message.message.details.expiration) * 1000) } countBackwards = { true }/>
```

Comment:
new Date instances could be called before the return call so it's only instantiated once.

Reply:
hmm what do you mean by this?

Reply:
I believe the idea is to just do `const priorTimestamp = new Date(...)` up above somewhere, then your elements would be something like `<SomeTimeAgo priorTimestamp countBackwards = { true }/>`